From 231b54b2d9c29130f12378ef039d898d917b4e3b Mon Sep 17 00:00:00 2001
From: Dinesh kumar <dinesh.v@adlinktech.com>
Date: Fri, 6 Feb 2026 12:09:34 +0530
Subject: [PATCH 08/19] Poweroff handler for bbm

---
 drivers/firmware/imx/sm-bbm.c | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/drivers/firmware/imx/sm-bbm.c b/drivers/firmware/imx/sm-bbm.c
index c400fe2e5b7f..21bc37485b9c 100644
--- a/drivers/firmware/imx/sm-bbm.c
+++ b/drivers/firmware/imx/sm-bbm.c
@@ -12,6 +12,7 @@
 #include <linux/scmi_protocol.h>
 #include <linux/scmi_nxp_protocol.h>
 #include <linux/suspend.h>
+#include <linux/reboot.h>
 
 #define DEBOUNCE_TIME		30
 #define REPEAT_INTERVAL		60
@@ -238,6 +239,18 @@ static int scmi_imx_bbm_rtc_init(struct scmi_device *sdev)
 								NULL, &bbnsm->nb);
 }
 
+static int bbm_poweroff(struct sys_off_data *data)
+{
+    struct scmi_imx_bbm *bbnsm = data->cb_data;
+    struct scmi_protocol_handle *ph = bbnsm->ph;
+    u32 state = 0;
+
+    printk("iMX95: notify System Manager to power off\n");
+
+    bbnsm->ops->shutdown_set(ph, &state);
+    return NOTIFY_DONE;
+}
+
 static int scmi_imx_bbm_probe(struct scmi_device *sdev)
 {
 	const struct scmi_handle *handle = sdev->handle;
@@ -275,6 +288,12 @@ static int scmi_imx_bbm_probe(struct scmi_device *sdev)
 		return ret;
 	}
 
+	ret = devm_register_power_off_handler(dev, bbm_poweroff, bbnsm);
+    if (ret) {
+        dev_err(dev, "poweroff register failed: %d\n", ret);
+        return ret;
+    }
+
 	return 0;
 }
 
-- 
2.43.0

