From 52c6d62d278aa381d688c1ab3900d09efaf16fe3 Mon Sep 17 00:00:00 2001
From: Craig Chen <craig.chen@adlinktech.com>
Date: Wed, 17 Jun 2026 12:47:29 +0000
Subject: [PATCH 2/2] lec-imx8mp-b1: fixed rtc alarm timer issue

Signed-off-by: M. Shangavelan <shangavelan.m@adlinktech.com>
---
 drivers/rtc/rtc-rx8130.c | 17 +++++++++--------
 1 file changed, 9 insertions(+), 8 deletions(-)

diff --git a/drivers/rtc/rtc-rx8130.c b/drivers/rtc/rtc-rx8130.c
index f45cac7c708c..fc21ca799840 100644
--- a/drivers/rtc/rtc-rx8130.c
+++ b/drivers/rtc/rtc-rx8130.c
@@ -482,18 +482,18 @@ static int rx8130_set_alarm(struct device *dev, struct rtc_wkalrm *t)
 	struct i2c_client *client = to_i2c_client(dev);
 	struct rx8130_data *rx8130 = dev_get_drvdata(dev);
 	u8 alarmvals[3];		//minute, hour, day
-	u8 extreg;				//extension register
-	u8 flagreg;				//flag register
+	u8 extreg;				//extension register(0x1c)
+	u8 flagreg;				//flag register(0x1d)
 	int err;
-	 
+ 
 	if (client->irq <= 0)
 		return -EINVAL;
-		
+
 	//get current extension register
 	err = rx8130_read_reg(client, RX8130_REG_EXT, &extreg);
 	if (err <0)
 		return err;
-	
+
 	//get current flag register
 	err = rx8130_read_reg(client, RX8130_REG_FLAG, &flagreg);
 	if (err <0)
@@ -512,7 +512,7 @@ static int rx8130_set_alarm(struct device *dev, struct rtc_wkalrm *t)
 		if (err)
 			return err;
 	}
-	
+
 	//write the new minute and hour values
 	//Note:assume minute and hour values will be enabled. Bit 7 of each of the
 	//     minute, hour, week/day register can be set which will "disable" the
@@ -539,7 +539,8 @@ static int rx8130_set_alarm(struct device *dev, struct rtc_wkalrm *t)
 	if (err)
 		return err;
 
-	//clear Alarm Flag
+	err = rx8130_read_reg(client,RX8130_REG_FLAG,&flagreg);
+	//clear alarm Flag
 	flagreg &= ~(RX8130_BIT_FLAG_AF | RX8130_BIT_FLAG_UF);
 	err = rx8130_write_reg(rx8130->client, RX8130_REG_FLAG, flagreg);
 	if (err)
@@ -572,7 +573,7 @@ static int rx8130_alarm_irq_enable(struct device *dev, unsigned int enabled)
 	u8 flagreg;
 	u8 ctrl;
 	int err;
-	
+
 	//get the current ctrl settings
 	ctrl = rx8130->ctrlreg;
 
-- 
2.34.1

